openedx_events/event_bus/__init__.py

```diff
@@ -183,3 +184,28 @@ def make_single_consumer(*, topic: str, group_id: str,
 def _reset_state(sender, **kwargs):  # pylint: disable=unused-argument
     """Reset caches when settings change during unit tests."""
     get_producer.cache_clear()
+
+
+def merge_publisher_configs(publisher_config_original, publisher_config_overrides):
+    """
+    Merge two EVENT_BUS_PRODUCER_CONFIG maps.
+
+    Arguments:
+        publisher_config_original: An EVENT_BUS_PRODUCER_CONFIG-structured map
+        publisher_config_overrides: An EVENT_BUS_PRODUCER_CONFIG-structured map
+
+    Returns:
+        A new EVENT_BUS_PRODUCER_CONFIG map created by combining the two maps. All event_type/topic pairs in
+        publisher_config_b are added to the publisher_config_a. If there is a conflict on whether a particular
+        event_type/topic pair is enabled, publisher_config_b wins out.
+    """
+    combined = copy.deepcopy(publisher_config_original)
```

Comment:
Note: use deepcopy here to prevent accidentally updating the original config object. Since it's a few layers deep, {**publisher_config_original} doesn't make a full deep copy.

Reply:
I presume you discovered this through a unit test, but if not, maybe ensure there is coverage for this?

Reply:
This is addressed in an existing unit test